opengrok-tools/src/main/python/opengrok_tools/scm/repository.py

```diff
@@ -75,6 +82,105 @@ def incoming(self):
         """
         Check if there are any incoming changes.
 
+        Return True if so, False otherwise.
+        """
+        if self.is_command_overridden(self.configured_commands, 'incoming'):
+            return self._run_custom_incoming_command(
+                self.listify(self.configured_commands['incoming'])
+            )
+        return self.incoming_check()
+
+    def incoming_check(self):
+        """
+        Check if there are any incoming changes.
+
         Return True if so, False otherwise.
         """
         return True
+
+    def _run_custom_sync_command(self, command):
```

Comment:
is this something that could be reused in the individual repository implementations ? same for _run_custom_incoming_command().

Reply:
I think it's possible, didn't do it for mercurial as it contains more complicated logic.

Reply:
I split it to a commit per repository type, could you review it again if that is ok? I don't have enough resources to test all of the repository types.